### PR TITLE
Add blogs

### PR DESCRIPTION
**Pull request by eleensmathew**
## Add Blogs

Added functionality to add blogs and fuxed the templates for seeing all blogs and seeing blog detail. Also implemented nested comments.

## Related Issue

Fixes # (https://github.com/ContriHUB/Youdemy/issues/2)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

**Comment by eleensmathew** [comment]
127 0 0 1_8000_blog_single_8_
blog_detail currently looks like this

**Comment by eleensmathew** [comment]
Have not made changes to Category, Popular articles and Archives

File: main/urls.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
     path('add_titles/<int:chapter_id>/',views.create_title_model_form,name='create_title_model_form'),
     path('add_question/<int:chapter_id>/',views.create_question,name='create_question'),
     path('add_chapter/<int:course_id>/',views.create_chapter,name='create_chapter'),
+    path('add_blog/',views.create_blog,name='create_blog'),
```
Comment (kunal2812) [inline main/urls.py]: Where is the option on frontend to create blog ? Can't see it anywhere

Comment (eleensmathew) [reply main/urls.py]: I'll work on it

Comment (kunal2812) [reply main/urls.py]: Also, ensure that I am able to replicate the blog as closely as possible to to static one using the functionality you implemented.

**Comment by eleensmathew** [comment]
I added link in the all blogs pags 
image
Added edit and delete blogs 
image
Slightly improved display of like and dislike button 
image
Changed display of email in comments to first name 
image

**Comment by eleensmathew** [comment]
I have made all the suggested changes. If you have any comments please let me know

**Comment by kunal2812** [comment]
After the blog is created when I click the blog card to view full blog I'm getting this error.
image

**Comment by eleensmathew** [comment]
That error is happening becaus the person is supposed to create a profile for himself. Right now the userprofile contains the bio so do you just want me to put some default value for all the fields of userprofile?

**Comment by kunal2812** [comment]
Yes those fields which are not asked during the signup can have the default values for now.

**Comment by eleensmathew** [comment]
Resolved the error. You have to make migrations and try creating a user from scratch to test it. 
image
The default values are seen above.

**Comment by kunal2812** [comment]
Issues:
1. If the user is not signed in don't show the functionality to add blog. Leads to this error.
image

2. Have you added paging to the blogs or are they appended in the same page itself ? If you are showing all blogs in same page then remove this from frontend.
image

3. All blog cards should be of same size as shown in #2. The blog heading should be shown in 2 lines and preview of blog content in 4 lines. Keep this consistent across all cards.
image

4. Fix css of blog details page footer
image

5. Currently blog content looks a bit congested. Break blog content into 3 parts - Introduction, Main Body, Conclusion and Place the blog thumbnail after the Introduction (refer static page for more details). 

6. If I comment via comment form then I am on the same page but when I comment on the thread then I'm taken back to Blog cards page (Same page should refresh).

7. Like Dislike not working
image

8. This takes me to "Home" page.
image

9. Time is in UTC. Change it to local time.
image

**Comment by eleensmathew** [comment]
changed the timezone 
image
added intro and conclusion 
127 0 0 1_8000_blog_1_
modified card display 
image
changed all the incorrect links and made add blog functionality only to registered users. Fixed like and dislike. Removed next page option as well.
However the css of the footer in consistent for me on all pages, so can you try clearing the cache and letting me know what happens?

**Review by kunal2812 — CHANGES_REQUESTED** [review]
> 3. The blog heading should be shown in 2 lines and preview of blog content in 4 lines. Keep this consistent across all cards.
>    image

This is not done yet.

Fix typo here
image

image
image

Check blogs page and blogs detail page. There is this extra white space.

Getting this error when I enter 0 tags when I create blog. And even though I was getting the error the blog was anyway still created. Either add the functionality to not add any tags or force the user to add atleast 1 tag (First option is preferable).
image

**Comment by eleensmathew** [comment]
do these cards look ok?
image
i made the cards all uniform height

**Comment by kunal2812** [comment]
> do these cards look ok? image i made the cards all uniform height

The blog card should consist of three sections: the Blog Heading, Blog Content, and the Read More part. Each of these sections should have a fixed height within the card, and they should all be aligned on the same line.

**Comment by eleensmathew** [comment]
i hope this is fine now ?
image

**Comment by kunal2812** [comment]
> i hope this is fine now ? image

Yes. But keep the First name of user and no. Of comments in line with "Read More" like before

**Comment by eleensmathew** [comment]
fixed the cards
image

fixed footer and header white space 
image
and fixed empty tags problem

**Comment by kunal2812** [comment]
When an unregistered user tries to like, dislike, or comment, they should be redirected to the sign-up page, as this action would otherwise result in an error.

Everything else appears to be okay.

**Comment by eleensmathew** [comment]
Can I display something like to view comments sign in? and to like and dislike this post sign in and i attach the link to home page if they click sign in?

**Comment by eleensmathew** [comment]
Something like this 
image

**Comment by kunal2812** [comment]
> Can I display something like to view comments sign in? and to like and dislike this post sign in and i attach the link to home page if they click sign in?

Yes you can do that.

**Comment by eleensmathew** [comment]
Made all the required changes. look like this now 
image

**Review by kunal2812 — APPROVED** [review]
LGTM

**Pull request merged**